src/sentry/integrations/helper.py

```diff
@@ -169,11 +184,13 @@ def _finish_pipeline(self, data):
                 name=data.get('name', self.provider.name),
             )
         else:
-            self.integration = Integration.objects.create(
+            self.integration, _ = Integration.objects.get_or_create(
                 provider=self.provider.id,
-                metadata=data.get('metadata', {}),
-                name=data.get('name', data['external_id']),
                 external_id=data['external_id'],
+                defaults={
+                    'metadata': data.get('metadata', {}),
```

Comment:
probably should update this when it already exists since this can include access tokens which may have been rotated